packages/ra-core/src/dataProvider/useDelete.ts

```diff
@@ -204,7 +204,7 @@ export const useDelete = <
                     'useDelete mutation requires a non-empty resource'
                 );
             }
-            if (!callTimeId) {
+            if (callTimeId === undefined || callTimeId === null) {
```

Comment:
We use

```suggestion
            if (callTimeId != null) {
```

in other parts of the core (more concise).